dataactvalidator/filestreaming/loadFile.py

```diff
@@ -1,53 +1,216 @@
+        {}
+    )
+    # de-dupe
+    data.drop_duplicates(subset=['object_class_code'], inplace=True)
+    # insert to db
+    LoaderUtils.insertDataframe(data, model.__table__.name, interface.engine)
 
-def loadSF133(filename):
-    """ Load SF133 files, set skipClear to True for second file """
-    LoaderUtils.loadCsv(filename,SF133,ValidatorValidationInterface(),{"ata":"allocation_transfer_agency","aid":"agency_identifier","availability_type_code":"availability_type_code","bpoa":"beginning_period_of_availa","epoa":"ending_period_of_availabil","main_account":"main_account_code","sub_account":"sub_account_code","fiscal_year":"fiscal_year","period":"period","line_num":"line","amount_summed":"amount"},{"allocation_transfer_agency":{"pad_to_length":3},"agency_identifier":{"pad_to_length":3},"main_account_code":{"pad_to_length":4},"sub_account_code":{"pad_to_length":3},"amount":{"strip_commas":True}})
 
 def loadProgramActivity(filename):
-    LoaderUtils.loadCsv(filename, ProgramActivity, ValidatorValidationInterface(), {"year":"budget_year","agency_id":"agency_id",
-        "alloc_id":"allocation_transfer_id","account":"account_number","pa_code":"program_activity_code","pa_name":"program_activity_name"},
-        {"program_activity_code":{"pad_to_length":4},"agency_id":{"pad_to_length":3},"account_number":{"pad_to_length":4},"allocation_transfer_id":{"pad_to_length":3}})
+    interface = ValidatorValidationInterface()
+    model = ProgramActivity
+
+    # for program activity, delete and replace values??
+    interface.session.query(model).delete()
+    interface.session.commit()
+
+    data = pd.read_csv(filename, dtype=str)
+    data = LoaderUtils.cleanData(
+        data,
+        model,
+        {"year": "budget_year",
+         "agency_id": "agency_id",
+         "alloc_id": "allocation_transfer_id",
+         "account": "account_number",
+         "pa_code": "program_activity_code",
+         "pa_name": "program_activity_name"},
+        {"program_activity_code": {"pad_to_length": 4},
+         "agency_id": {"pad_to_length": 3},
+         "account_number": {"pad_to_length": 4},
+         "allocation_transfer_id": {"pad_to_length": 3}}
+    )
+    # because we're only loading a subset of program activity info,
+    # there will be duplicate records in the dataframe. this is ok,
+    # but need to de-duped before the db load.
+    data.drop_duplicates(inplace=True)
+    # insert to db
+    LoaderUtils.insertDataframe(data, model.__table__.name, interface.engine)
+
+
+def loadSF133(filename, fiscal_year, fiscal_period, force_load=False):
+    interface = ValidatorValidationInterface()
+    model = SF133
+
+    existing_records = interface.session.query(model).filter(
+        and_(model.fiscal_year == fiscal_year, model.period == fiscal_period))
+    if force_load:
+        # force a reload of this period's current data
+        print('Force SF 133 load: deleting existing records for {} {}'.format(
```

Comment:
The strategy I learned under @vrajmohan and @jmcarp's excellent tutelage is to create a module-level `logger` and configure everything at the app's entry points. It'd look something like this:

```python
# my/module/here.py
import logging

logger = logging.getLogger(__name_)

def some_func():
    logger.info("Some Func called")    # or debug, warning, etc.

# run.py
import logging

if __name__ == '__main_':
    logging.basicConfig()    # Can be configured with threshold level here
    runProgram()
```

A better long-term strategy would be to include config information in the project's settings so it can be configured without modifying code -- Django does this, but Flask's intentionally minimal.